sys/include/fido2/ctap/ctap_mem.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    fido2_ctap_mem FIDO2 CTAP flash
+ * @ingroup     fido2_ctap
+ * @brief       FIDO2 CTAP flash memory helper
+ *
+ * @{
+ *
+ * @file
+ * @brief       Definitions for CTAP flash memory helper functions
+ *
+ * @author      Nils Ollrogge <nils.ollrogge@fu-berlin.de>
+ */
+
+#ifndef FIDO2_CTAP_CTAP_MEM_H
+#define FIDO2_CTAP_CTAP_MEM_H
+
+#include <stdint.h>
+
+#include "fido2/ctap/ctap.h"
+#include "periph/flashpage.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Start page for storing resident keys
+ */
+#define CTAP_FLASH_RK_START_PAGE 15U
```

Comment:
I only store the `ctap_state_t` struct in a separate flashpage. The `ctap_resident_key_t` structs are stored in the same flashpage until it is full. 

I don't need guarantees about how many flash pages there will be, I just need to know how many there are in order to limit the maximum amount of credentials that can be stored.

No, wear-leveling is not the goal yet.

Reply:
`flashpage_first_free` looks very promising ! This would require a specific function for each CPU though I believe ?

Reply:
`flashpage_first_free` seems nice but waiting for such a feature is not needed. Especially the memory abstraction of this PR will get  follow-up work anyway. Hence, I'd vote for some mechanism that reserves "a couple of" flashpages at the end of the memory and documents that behavior properly. As far as I see it, the nrf52 is currently the only capable platform anyway.

@benpicco would you mind to have a look if your comments were addressed properly?

Reply:
@benpicco ?

Reply:
Fine  with me, I see you want the protocol in but the storage backend could need some work.
Please add a note somewhere that this can lead to surprise firmware corruption if `FLASHPAGE_SIZE` is not right.

Also please make it configurable

How about

```C
#ifndef CTAP_FLASH_RK_START_PAGE
#define CTAP_FLASH_RK_START_PAGE (FLASHPAGE_NUMOF - 4)
#endif
```